src/windows.rs

```diff
@@ -86,238 +121,320 @@ impl crate::Brightness for Brightness {
     }
 
     async fn get(&self) -> Result<u32, Error> {
-        let ioctl_query = ioctl_query_supported_brightness(self);
-        match ioctl_query {
-            Ok(_) => Ok(ioctl_query_display_brightness(self)?),
-            Err(e) if is_not_found(&e) => {
-                Ok(ddcci_get_monitor_brightness(self)?.get_current_percentage())
-            }
-            Err(e) => Err(SysError::IoctlQuerySupportedBrightnessFailed {
-                device_name: self.device_name.clone(),
-                source: e,
-            }
-            .into()),
-        }
+        Ok(if self.is_internal() {
+            ioctl_query_display_brightness(self)?
+        } else {
+            ddcci_get_monitor_brightness(self)?.get_current_percentage()
+        })
     }
 
     async fn set(&mut self, percentage: u32) -> Result<(), Error> {
-        let ioctl_query = ioctl_query_supported_brightness(self);
-        match ioctl_query {
-            Ok(levels) => {
-                let new_value = levels.get_nearest(percentage);
-                Ok(ioctl_set_display_brightness(self, new_value)?)
-            }
-            Err(e) if is_not_found(&e) => {
-                let new_value =
-                    ddcci_get_monitor_brightness(self)?.percentage_to_current(percentage);
-                Ok(ddcci_set_monitor_brightness(self, new_value)?)
-            }
-            Err(e) => Err(SysError::IoctlQuerySupportedBrightnessFailed {
-                device_name: self.device_name.clone(),
-                source: e,
-            }
-            .into()),
-        }
+        Ok(if self.is_internal() {
+            let supported = ioctl_query_supported_brightness(self)?;
+            let new_value = supported.get_nearest(percentage);
+            ioctl_set_display_brightness(self, new_value)?;
+        } else {
+            let current = ddcci_get_monitor_brightness(self)?;
+            let new_value = current.percentage_to_current(percentage);
+            ddcci_set_monitor_brightness(self, new_value)?;
+        })
     }
 }
 
 pub fn brightness_devices() -> impl Stream<Item = Result<Brightness, SysError>> {
```

Comment:
I have broken up parts of this into their own functions; it was becoming difficult to read because of both its length and depth of nested blocks (vs the line length limit).
This has also created room for me to add more detailed rustdoc style comments to each of the new functions.
Hopefully this is ok?

Reply:
Shorter and simpler functions and documentation are always welcome. Thank you!